packages/p2p-v2/src/components/FlyoutMenu/__test__/FlyoutMenu.spec.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import FlyoutMenu from '../FlyoutMenu';
+
+const flyoutItems = ['item1', 'item2', 'item3'];
+
+describe('FlyoutMenu', () => {
+    it('it should render the flyout menu correctly', () => {
+        render(<FlyoutMenu listItems={flyoutItems} renderIcon={() => 'MockIcCashierVerticalEllipsis'} />);
+        expect(screen.getByText('MockIcCashierVerticalEllipsis')).toBeInTheDocument();
+    });
+    it('it should display the menu items when the icon is clicked', () => {
+        render(<FlyoutMenu listItems={flyoutItems} />);
+        userEvent.click(screen.getByTestId('dt_p2p_v2_flyout_toggle'));
+        flyoutItems.forEach(item => {
+            expect(screen.getByText(item)).toBeInTheDocument();
+        });
+    });
+    it('should call the usecallback callback, when the parent is clicked', () => {
+        render(
+            <div data-testid='dt_p2p_v2_flyout_parent'>
+                <FlyoutMenu listItems={flyoutItems} />
+            </div>
+        );
+        userEvent.click(screen.getByTestId('dt_p2p_v2_flyout_toggle'));
+        expect(screen.queryByText(flyoutItems[0])).toBeInTheDocument();
+        userEvent.click(screen.getByTestId('dt_p2p_v2_flyout_parent'));
+        expect(screen.queryByText(flyoutItems[0])).not.toBeInTheDocument();
```

Comment:
so shouldn't it test for some fn to have been called instead according to the test case description?

Reply:
Hi @nada-deriv, As seen in the screenshot attached, the `callback function` can't be mocked, so there is no way of telling whether it's been called or not. It's called directly inside of the `useOnClickOutside` hook. Maybe I'll update the description of the test case.

<img width="404" alt="useCallBack" src="https://github.com/binary-com/deriv-app/assets/104334373/c879cf6e-f8ee-4efc-9880-437cb744a10a">

Reply:
ahh.. i think then may be the description can be changed.